### PR TITLE
v3.1.0 Matrix API

### PR DESCRIPTION
**Pull request by corypisano**
Add Matrix API support
Add `events` to update trip results
Add config for base api path

### CODE REVIEW

**Comment by corypisano** [comment]
https://app.clubhouse.io/radarlabs/story/6528/return-events-in-js-sdk-patch-trips

File: src/api/matrix.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import Http from '../http';
+import Navigator from '../navigator';
+
+class Matrix {
+  static async getMatrixDistances(routingOptions={}) {
```
Comment (kochis) [inline src/api/matrix.js:5]: Should this just be in `src/api/routing.js`? Not sure we need a separate file for matrix

Comment (corypisano) [reply src/api/matrix.js:5]: ah yea i'll move it

**Review by andlum — APPROVED** [review]
The changes look good to me; I didn't know how to QA it but I mostly just cross-referenced the [iOS docs](https://radar.com/documentation/sdk/ios#matrix) for the Matrix API support

**Pull request merged**